trieste/bayesian_optimizer.py

```diff
@@ -182,7 +183,20 @@ def optimize(
         num_steps: int,
         datasets: Mapping[str, Dataset],
         model_specs: Mapping[str, ModelSpec],
-        acquisition_rule: AcquisitionRule[S, SP],
+        acquisition_rule: AcquisitionRule[TensorType, SP],
+        *,
+        track_state: bool = True,
+        fit_intial_model: bool = True,
+    ) -> OptimizationResult[object]:
```

Comment:
@uri-granta I've used `object` here instead of `None` because mypy can't differentiate between `TensorType` and `State[...]`, so the return types clash if it's `None`. I think this should be OK cos `None` and `object` have (I think) the same interface

Reply:
Ugh! Maybe put a comment?

Reply:
done